x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -100,20 +104,50 @@ export function trainedModelsRoutes({ router, routeGuard }: RouteInitialization)
                   ...Object.values(modelDeploymentsMap).flat(),
                 ])
               );
+              const modelsClient = modelsProvider(client);
 
-              const pipelinesResponse = await modelsProvider(client).getModelsPipelines(
-                modelIdsAndAliases
+              const modelsPipelinesAndIndices = await Promise.all(
+                modelIdsAndAliases.map(async (modelIdAndAlias) => {
+                  return {
+                    modelIdAndAlias,
+                    result: await modelsClient.getModelsPipelinesAndIndicesMap(modelIdAndAlias),
```

Comment:
⚠️  I checked the `getModelsPipelinesAndIndicesMap` and it seems like it does some unnecessary work. For instance, it fetches the pipeline definition one more time. We should optimize it to reduce the number of API calls that happen in this endpoint.

Reply:
This `getModelsPipelinesAndIndicesMap` is a replacement/extended version of getModelsPipelines in previous versions (see [usage in 8.7](https://github.com/elastic/kibana/blob/adf73ca765fc5a218bf0b2b903a72aae54acf6d1/x-pack/plugins/ml/server/routes/trained_models.ts#L104C47-L104C88)). This would not be called again if the Models map view is not opened in the Trained models page. Currently, refactoring it to reduce this call requires lots of churn in the analytics map end point on both UI and server. I can look into potentially optiziming this more in a follow up.

Reply:
Created an issue for it here https://github.com/elastic/kibana/issues/166735